### PR TITLE
Implement support for @objc(name) on enum cases

### PR DESCRIPTION
**Pull request by lilyball**

### CODE REVIEW

**Comment by kballard** [comment]
PR updated

@jrose-apple I ended up going with your original suggested error text. It is a bit wordy, but it does get across the desired message.

**Comment by jrose-apple** [comment]
That is my only comment so I'll merge anyway and just fix it afterwards. :-)

**Comment by kballard** [comment]
I can push an amended commit easily enough.

**Comment by jrose-apple** [comment]
Oh, well if you're around then okay!

**Comment by kballard** [comment]
PR updated.

**Comment by jrose-apple** [comment]
Oh, I guess this should get release-noted too.

**Comment by kballard** [comment]
Oh right. I'll add a commit for that.

**Comment by kballard** [comment]
Hrm, looks like we forgot to add a release note for supporting `@objc(EnumName)` on enums. Tell you what, go ahead and merge this and I'll submit a separate PR to add both notes. I'd have to rebase this PR anyway to avoid a merge conflict on CHANGELOG if I were to add it right now.

**Comment by jrose-apple** [comment]
Seems reasonable. Thanks!

**Comment by kballard** [comment]
Submitted the CHANGELOG change as #900.

**Comment by lattner** [comment]
Awesome, thanks Kevin!
